source/common/common/utility.h

```diff
@@ -768,13 +769,168 @@ class InlineStorage : public NonCopyable {
   }
 };
 
+// Utility functions and stateful representation to efficiently encode and
```

Comment:
Note to reviewers: this code was moved from its prior location in `Stats::SymbolTable::Encoding`, to its own standalone helper class NumericEncoding. The implementations are inlined rather than in a C++ file to avoid a performance degradation measured in test/common/stats/symbol_table_speed_test.cc. Previously, with Stats::SymbolTable::Encoding's implementation visible to the compiler at its call-site, the compiler can optimize better than if it only sees the API in the header.